nav2_planner/src/planner_server.cpp

```diff
@@ -407,7 +409,14 @@ void PlannerServer::computePlanThroughPoses()
       if (i == 0) {
         curr_start = start;
       } else {
-        curr_start = goal->goals[i - 1];
+        if (allow_path_end_pose_deviation_) {
+          // pick the end of the last planning task as the start for the next one
+          curr_start = concat_path.poses.back();
```

Comment:
Curious: test with Smac Hybrid, Lattice, NavFn. Does the `curr_start == goal->goals[i - 1]`? I believe it might. If so, we don't even necessarily need to make this a parameter nor a `if/else` condition.

Reply:
I tested with this planners and it seems to work (hybrid and lattice seem to be harder to use in close proximity of obstacles in general with the default configuration though):
NavFn:
![Screenshot from 2023-11-06 07-56-09](https://github.com/ros-planning/navigation2/assets/71234974/094197f8-0bf0-48af-aa87-59c92985b6cc)
Hybrid:
![Screenshot from 2023-11-06 08-09-23](https://github.com/ros-planning/navigation2/assets/71234974/5db6150b-ada6-45f5-95bc-c787c42fdbea)
Lattice:
![Screenshot from 2023-11-06 08-30-42](https://github.com/ros-planning/navigation2/assets/71234974/d60d1ac6-280f-442d-adb0-0a32e1b62616)

>  We could just do this as the default behavior and add to our migration guide that it supports deviation up to the planner's tolerance.

I agree this should be the default behavior: if there's a user defined tolerance navigation should succeed as long as the planner can get within that tolerance to goals, I can't think of a good example where an user may want `compute_path_though_poses` to fail where that is possible. They can always set a very low tolerance if having goals in non-traversable space is a critical condition

Reply:
Lattice shows a strange discontinuity - its hard for me to tell from the resolution of the picture of that's a problem or not. Is that within a single costmap cell (and just goal approximation within a cell issues)? Does that happen when the approximate option is disabled?

Reply:
@pepisg any update?

Reply:
Hey! I havent had time to keep reviewing this, specially for writing the new tests for the navigate through poses action (planner tests seem to test the functions on planner server directly, so it looked like a complete new testing module to test the actions would have to be written).

I will try to make time for this in the coming weeks, sorry for the delay